### PR TITLE
Add more error handling in writeBytes.

### PR DESCRIPTION
**Pull request by hiddenalpha**
- Throw NullPointerException (in place of crashing) if caller passes null buffer.
- Pass-through OutOfMemoryError if GetByteArrayElements() fails to alloc memory.

### CODE REVIEW

**Comment by tresf** [comment]
> Pass-through OutOfMemoryError if GetByteArrayElements() fails to alloc memory.

Can you explain where the `OutOfMemoryError` is thrown?

> Throw NullPointerException (in place of crashing) if caller passes null buffer.

Hmm... Should we wrapped exceptions which are historically non-checked?

**Comment by hiddenalpha** [comment]
Will have a look as soon I find some time. Thx for feedback so far.

**Comment by tresf** [comment]
> > Hmm... Should we wrap exceptions which are historically non-checked?

> Will have a look as soon I find some time. Thx for feedback so far.

Whoops, we don't.  We only catch `IOException`, so we're still going to throw unchecked exceptions, which I think is what we want, per:

https://github.com/java-native/jssc/blob/ea2ade2b3cbbc417ef9dfd97922a77fde41b911a/src/main/java/jssc/SerialPort.java#L414-L416

I think unchecked exceptions are fine for now (they're certainly better than a JVM crash).  If users complain about this we, can modify the the catch statement.

**Comment by tresf** [comment]
@hiddenalpha do you believe we should add more unit tests for these?  Furthemore, have you had time to evaluate the safety of the windows code too?

**Comment by hiddenalpha** [comment]
> do you believe we should add more unit tests for these?

I think writing unit tests may be challenging for some of those cases. Especially edge-cases in the native code. I think best is to add a test when we see a good opportunity and if it can be written in a simple (*as in understandable/maintainable*) way.

> have you had time to evaluate the safety of the windows code too?

Did not look at the windows code at all yet. Will keep it in mind when I've a free time slot.

File: src/test/java/jssc/SerialNativeInterfaceTest.java [hunk 1]
```diff
@@ -47,4 +55,16 @@ public void reportsWriteErrorsAsIOException() throws Exception {
...
+    public void throwsNpeIfPassedBufferIsNull() throws Exception {
+        long fd = 1;
+        byte[] buf = null;
+        try{
+            testTarget.writeBytes(fd, buf);
+            fail("Where is the exception?");
+        }catch( NullPointerException ex ){
+            assertTrue(ex.getMessage().contains("buffer"));
+        }
+    }
```
Comment (tresf) [inline src/test/java/jssc/SerialNativeInterfaceTest.java:68]: Should we copy the `expected =` signature in case we change the messaging later?

```suggestion
    @Test(expected = java.lang.NullPointerException.class)
    public void throwsNpeIfPassedBufferIsNull() throws Exception {
        long fd = 1;
        byte[] buf = null;
        testTarget.writeBytes(fd, buf);
    }
```

**Comment by tresf** [comment]
This can be tested in its current form using `mvn -DskipTests install && mvn` until rebased against master (specifically #51, #151).

It looks good to me, but I've left a question about the unit test, since matching an exception string can have unobvious test failures in the even this ever gets changed.

**Comment by tresf** [comment]
When you have time, this PR needs to be rebased and there's still one comment remaining.  I've love to see it merged. :)
